theatre/studio/src/panels/DetailPanel/propEditors/DeterminePropEditor.tsx

```diff
@@ -1,115 +1,140 @@
   PropTypeConfig,
   {type: K}
 >
 
-type IPropEditorByPropType = {
-  [K in PropTypeConfig['type']]: React.FC<{
-    obj: SheetObject
-    pointerToProp: Pointer<PropConfigByType<K>['valueType']>
-    propConfig: PropConfigByType<K>
-    visualIndentation: number
-  }>
+type ISimplePropEditorByPropType = {
+  [K in PropTypeConfig_AllSimples['type']]: ISimplePropEditorVFC<
+    PropConfigByType<K>
+  >
 }
 
-const propEditorByPropType: IPropEditorByPropType = {
-  compound: CompoundPropEditor,
+const simplePropEditorByPropType: ISimplePropEditorByPropType = {
   number: NumberPropEditor,
   string: StringPropEditor,
-  enum: () => <></>,
   boolean: BooleanPropEditor,
   stringLiteral: StringLiteralPropEditor,
   rgba: RgbaPropEditor,
 }
 
-export type IEditablePropertyProps<K extends PropTypeConfig['type']> = {
+type IDetailEditablePropertyProps<K extends PropTypeConfig['type']> = {
   obj: SheetObject
   pointerToProp: Pointer<PropConfigByType<K>['valueType']>
   propConfig: PropConfigByType<K>
 }
 
-type IDeterminePropEditorProps<K extends PropTypeConfig['type']> =
-  IEditablePropertyProps<K> & {
+type IDetailDeterminePropEditorProps<K extends PropTypeConfig['type']> =
+  IDetailEditablePropertyProps<K> & {
     visualIndentation: number
   }
 
-const DeterminePropEditor: React.FC<
-  IDeterminePropEditorProps<PropTypeConfig['type']>
+type IKeyframeDeterminePropEditorProps<K extends PropTypeConfig['type']> = {
+  editingTools: IEditingTools<PropConfigByType<K>['valueType']>
+  propConfig: PropConfigByType<K>
+  keyframeValue: PropConfigByType<K>['valueType']
+  displayLabel?: string
+}
+
+export const DetailDeterminePropEditor: React.VFC<
+  IDetailDeterminePropEditorProps<PropTypeConfig['type']>
 > = (p) => {
   const propConfig =
     p.propConfig ?? getPropTypeByPointer(p.pointerToProp, p.obj)
 
-  const PropEditor = propEditorByPropType[propConfig.type]
-
-  return (
-    <PropEditor
-      obj={p.obj}
-      visualIndentation={p.visualIndentation}
-      // @ts-expect-error This is fine
-      pointerToProp={p.pointerToProp}
-      // @ts-expect-error This is fine
-      propConfig={propConfig}
-    />
-  )
+  if (propConfig.type === 'compound') {
+    return (
+      <DetailCompoundPropEditor
+        obj={p.obj}
+        visualIndentation={p.visualIndentation}
+        pointerToProp={p.pointerToProp}
+        propConfig={propConfig}
+      />
+    )
+  } else if (propConfig.type === 'enum') {
+    // hmm: document how enum gets put into ui (through label?)
+    return <></>
+  } else {
+    const PropEditor = simplePropEditorByPropType[propConfig.type]
+
+    return (
+      <DetailSimplePropEditor
+        SimpleEditorComponent={PropEditor}
+        obj={p.obj}
+        visualIndentation={p.visualIndentation}
+        pointerToProp={p.pointerToProp}
+        propConfig={propConfig}
+      />
+    )
+  }
 }
 
-export default DeterminePropEditor
+const KeyframePropEditorContainer = styled.div`
```

Comment:
Should from here on be moved to a new file? Why does it share the file with `DetailDeterminePropEditor`?